src/runtime/context.coffee

```diff
@@ -129,11 +129,22 @@ module.exports.Context = class Context
   # Recursive function that will grab nested support library localId results
   supportLibraryLocalIds: (lib, localIdResults) ->
     # Set library identifier name as the key and the object of localIds with their results as the value
-    localIdResults[lib.library.source.library.identifier.id] = lib.localId_context
+    # if it already exists then we need to merge the results instead of overwriting
+    if localIdResults[lib.library.source.library.identifier.id]?
+      @mergeLibraryLocalIdResults localIdResults, lib.library.source.library.identifier.id, lib.localId_context
+    else
+      localIdResults[lib.library.source.library.identifier.id] = lib.localId_context
     # Iterate over any support libraries in the current support library
     for supportLibName, supportLib of lib.library_context
       @supportLibraryLocalIds supportLib, localIdResults
 
+  mergeLibraryLocalIdResults: (localIdResults, libraryId, libraryResults) ->
+    for localId, localIdResult of libraryResults
+      existingResult = localIdResults[libraryId][localId]
+      # overwite this localid result if the existing result is "falsey". future work could track all results for each localid
```

Comment:
I'm not sure I understand the use case.  Why are `false` and `[]` candidates to be overwritten, but other values aren't?  In the other cases (when it's set to something other than `false` or `[]`), why do we throw out the newer results, preferring the older ones?

I'm not saying this is wrong, I'd just like to understand the reasoning for this approach.

Reply:
Summarizing Slack conversation:

Currently we use the results for coverage in bonnie which considers a covered clause to be something that is "truthy". So we want to overwrite anything that is "falsey". This uses the same logic for overwriting that is used above in `setLocalIdWithResult`. I will update the comments to explain this better.